src/control/jobs/control_jobs.c

```diff
@@ -580,6 +580,50 @@ static int32_t dt_control_flip_images_job_run(dt_job_t *job)
   dt_control_queue_redraw_center();
   return 0;
 }
+static int32_t dt_control_monochrome_images_job_run(dt_job_t *job)
+{
+  dt_control_image_enumerator_t *params = dt_control_job_get_params(job);
+  const int32_t mode = params->flag;
+  GList *t = params->index;
+  const guint total = g_list_length(t);
+  char message[512] = { 0 };
+
+  dt_undo_start_group(darktable.undo, DT_UNDO_LT_HISTORY);
```

Comment:
LT_HISTORY does not contains tags, so the added monochrome tags won't be removed by undo. I think you need:
`DT_UNDO_LT_HISTORY | DT_UNDO_TAGS`